src/unicycle_2d/ignition.cpp

```diff
@@ -106,8 +110,10 @@ void Ignition::onInit()
     &Ignition::setPoseDeprecatedServiceCallback,
     this);
 
+  // TODO(swilliams) Should this be executed every time optimizer.reset() is called, or only once ever?
```

Comment:
Does everyone agree this is the desired logic?

Scenario 1 - publish_on_startup: false
Startup: No transaction is published
SetPose called: Optimizer is reset, Transaction with provided pose is sent
Manual Optimizer reset: No transaction is published

Scenario 2 - publish_on_startup: true
Startup: Initial transaction is published
SetPose called: Optimizer is reset, Transaction with provided pose is sent
Manual Optimizer reset: No transaction is published  <-- I guess this is really the case I'm curious about?

Reply:
IMHO, "Manual Optimizer reset: No transaction is published" sounds like the correct thing to do in both cases.

For me `publish_on_startup` should only control the "Startup" case. Then, the user can use "SetPose" to ignite the smoother, and "reset" to stop it. At least that's my understanding. This will allow to start / stop it from outside, as we can currently do with `r_l`.